forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/rest/FilterRestResource.java

```diff
@@ -1,48 +1,29 @@
 package org.camunda.bpm.extension.hooks.rest;
 
-import org.camunda.bpm.engine.rest.dto.CountResultDto;
-import org.camunda.bpm.engine.rest.dto.runtime.FilterDto;
-import org.camunda.bpm.engine.rest.hal.Hal;
-import org.springframework.hateoas.EntityModel;
 import javax.ws.rs.Consumes;
-import javax.ws.rs.Path;
-import javax.ws.rs.GET;
 import javax.ws.rs.POST;
-import javax.ws.rs.PathParam;
+import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.Context;
 import javax.ws.rs.core.MediaType;
-import javax.ws.rs.core.UriInfo;
 import javax.ws.rs.core.Request;
 
-import java.util.List;
+import org.camunda.bpm.engine.rest.hal.Hal;
+import org.camunda.bpm.extension.hooks.rest.dto.TaskQueryDto;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
 
 @Produces(MediaType.APPLICATION_JSON)
 public interface FilterRestResource extends RestResource {
 
     String PATH = "/filter";
 
-    @GET
-    @Produces(MediaType.APPLICATION_JSON)
-    List<FilterDto> getFilters(@Context UriInfo uriInfo, @QueryParam("itemCount") Boolean itemCount,
-                                          @QueryParam("firstResult") Integer firstResult, @QueryParam("maxResults") Integer maxResults);
-
-    @GET
-    @Path("/{id}/list")
-    @Produces({MediaType.APPLICATION_JSON, Hal.APPLICATION_HAL_JSON})
-    Object executeList(@Context Request request, @QueryParam("firstResult") Integer firstResult, @QueryParam("maxResults") Integer maxResults, @PathParam("id") String id);
-
     @POST
-    @Path("/{id}/list")
```

Comment:
Can we do this is in a way to not break existing implementations if we merge the code. I mean keep existing endpoints for filters there and mark as deprecated. Then create a new resource for new implementation? If the name `/filters/tasks` is causing issue, let's create another name like `task-filters`. What do you think ?

Reply:
I agree, if it's the new implementation for existing endpoints a new versioned endpoint would be ideal, `/v2/filters/tasks`. I am not sure if it's the case here.

Reply:
Better to create a v2 resource and add the new endpoints in that. That was the idea when we designed this pattern. Also, mark the v1 resource as deprecated would be a good idea.
NOTE: this need to be done only if there is a drastic change in the existing resource / implementation. If the purpose is just to create a new api which does something else then go with v1 itself.

Reply:
It's not really v2 of existing endpoints, rather a new implementation by not using any of the underlying filter implementation. We could go with v2 though, but doesn't really align with typical versioning.

Reply:
Yeah I agree, I think we can go with v1 and add a new resource for new implementation. Also existing implementation can be deprecated.